packages/core-ethereum/src/ethereum.ts

```diff
@@ -94,7 +101,23 @@ export async function createChainWrapper(
     provider
   ) as HoprNetworkRegistry
 
-  const genesisBlock = parseInt(hoprChannelsDeployment.blockNumber)
+  //getGenesisBlock, taking the earlier deployment block between the channel and network Registery
+  let channelDeployBlockNumber: any
+  if (Object.keys(hoprChannelsDeployment).some((key) => key === 'receipt')) {
+    channelDeployBlockNumber = (hoprChannelsDeployment as Deployment).receipt.blockNumber
+  } else {
+    channelDeployBlockNumber = (hoprChannelsDeployment as ContractData).blockNumber
+  }
```

Comment:
how come we expect a difference here?

Reply:
For legacy deployment files, it's possible that deployment file won't be overwritten, and thus no `receipt`

Reply:
👍 let's add a comment on that